examples/counter/containers/App.js

```diff
@@ -1,16 +1,11 @@
 import React, { Component } from 'react';
-import CounterApp from './CounterApp';
-import { createStore, applyMiddleware, combineReducers } from 'redux';
-import thunk from 'redux-thunk';
 import { Provider } from 'react-redux';
-import * as reducers from '../reducers';
-
-const createStoreWithMiddleware = applyMiddleware(thunk)(createStore);
-const reducer = combineReducers(reducers);
-const store = createStoreWithMiddleware(reducer);
+import CounterApp from './CounterApp';
+import createCounterStore from '../store/createCounterStore';
 
 export default class App extends Component {
   render() {
+    const store = createCounterStore(this.props.initialState);
```

Comment:
The store should not be created inside the `render` function, otherwise you may have performance and correctness problems. See https://github.com/gaearon/redux/issues/287.